targets/unity-v2/source/Authentication/PlayFabAuthService.cs

```diff
@@ -0,0 +1,254 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using PlayFab.Authentication.Strategies;
+using PlayFab.ClientModels;
+using PlayFab.Internal;
+using UnityEngine;
+
+namespace PlayFab
+{
+    /// <summary>
+    /// Supported Authentication types
+    /// Note: Add types to there to support more AuthTypes
+    /// See - https://api.playfab.com/documentation/client#Authentication
+    /// </summary>
+    public enum AuthTypes
+    {
+        None = 0,
+        Silent,
+        EmailPassword,
+        UsernamePassword,
+        GooglePlayGames,
+        Facebook,
+        FacebookInstantGames,
+        GameCenter,
+        Kongregate,
+        NintendoSwitch,
+        XBoxLive,
+        PlayStation,
+        OpenId,
+        WindowsHello,
+        Twitch,
+        Steam
+    }
+
+    public class PlayFabAuthService
+    {
+        // Events to subscribe to for this service
+        public delegate void DisplayAuthenticationEvent();
+        public event DisplayAuthenticationEvent OnDisplayAuthentication;
+
+        public delegate void LoginSuccessEvent(LoginResult success);
+        public event LoginSuccessEvent OnLoginSuccess;
+
+        public delegate void PlayFabErrorEvent(PlayFabError error);
+        public event PlayFabErrorEvent OnPlayFabError;
+
+        public delegate void PlayFabLink(AuthTypes authType, PlayFabError error = null);
+        public event PlayFabLink OnPlayFabLink;
+        public event PlayFabLink OnPlayFabUnlink;
+
+        public string Email;
+        public string Username;
+        public string Password;
+        public GetPlayerCombinedInfoRequestParams InfoRequestParams;
+        public bool ForceLink;
+
+        public PlayFabAuthenticationContext AuthenticationContext { get; private set; }
+
+        private const string _LoginRememberKey = "PlayFabLoginRemember";
```

Comment:
Can you put a comment that these pref-keys are incompatible with split-screen/multi-instance play?
There's no good solution to that, because it's dependent on information we can't have.
So just put a comment here saying there's an issue to be fixed with split-screen/multi-instance play regarding these pref-keys.